src/web/app/components/dialogs/CodeGenerator/svgOperation.ts

```diff
@@ -0,0 +1,225 @@
+    text.setAttribute('font-postscript', font.postscriptName);
+    text.removeAttribute('style');
+  });
+
+  return svgElement;
+}
+
+function getTranslateValues(transform: string): { x: number; y: number } {
+  // Regular expression to match the translate values
+  const match = transform.match(/translate\(([^,]+),?\s*([^)]+)?\)/);
+
+  // If there's a match, extract x and y values
+  if (match) {
+    const x = parseFloat(match[1]); // First captured group is x
+    const y = match[2] ? parseFloat(match[2]) : 0; // Second group is y, default to 0 if missing
+
+    return { x, y };
+  }
+
+  // Default to 0 if no translate function is found
+  return { x: 0, y: 0 };
+}
+
+/* Barcode */
+export async function importBarcodeSvgElement(
+  svgElement: SVGElement,
+  isInvert = false
+): Promise<void> {
+  const batchCmd = new history.BatchCommand('Import Barcode');
+  const doc = preProcessTextTag(svgElement);
+  const group = Array.of<SVGElement>();
+  const gElements = doc.querySelectorAll('g');
+  const fontObj = await getFontObj(
+    fontFuncs.getFontOfPostscriptName(doc.querySelector('text')?.getAttribute('font-postscript'))
+  );
+
+  gElements.forEach((g) => {
+    const transform = getTranslateValues(g.getAttribute('transform'));
+
+    g.querySelectorAll('rect').forEach((rect) => {
+      const { x, y, width, height } = rect.getBBox();
+
+      group.push(
+        svgCanvas.addSvgElementFromJson({
+          element: 'rect',
+          curStyles: false,
+          attr: {
+            x: x + transform.x,
+            y: y + transform.y,
+            width,
+            height,
+            stroke: '#000',
+            fill: 'black',
+            opacity: 1,
+            'fill-opacity': 1,
+            id: svgCanvas.getNextId(),
+          },
+        })
+      );
+    });
+
+    g.querySelectorAll('text').forEach((text) => {
+      const { textContent } = text;
+
+      if (!textContent) {
+        return;
+      }
+
+      const tspan = document.createElementNS(SVG_NS, 'tspan');
+
+      tspan.setAttribute('x', `${Number.parseFloat(text.getAttribute('x')) + transform.x}`);
+      tspan.setAttribute('y', `${Number.parseFloat(text.getAttribute('y')) + transform.y}`);
+      tspan.textContent = textContent;
+      text.textContent = '';
+      text.appendChild(tspan);
+
+      const { d } = convertTextToPathByFontkit(text, fontObj);
+
+      group.push(
+        svgCanvas.addSvgElementFromJson({
+          element: 'path',
+          curStyles: true,
+          attr: {
+            d,
+            fill: 'black',
+            opacity: 1,
+            'fill-opacity': 1,
+            id: svgCanvas.getNextId(),
+          },
+        })
+      );
+    });
+  });
+
+  group.forEach((element) => {
+    batchCmd.addSubCommand(new history.InsertElementCommand(element));
+  });
+
+  doc.remove();
+
+  svgCanvas.selectOnly(group);
+  batchCmd.addSubCommand(svgCanvas.groupSelectedElements(true));
```

Comment:
May need to check if svgCanvas.groupSelectedElements(true) is `null` or add filter in batchCmd